chainlibpy/grpc_client.py

```diff
@@ -47,8 +47,17 @@
 class GrpcClient:
     DEFAULT_GAS_LIMIT = 200000
 
-    def __init__(self, wallet: Wallet, chain_id: str, grpc_endpoint: str) -> None:
-        channel = insecure_channel(grpc_endpoint)
+    def __init__(
+        self,
+        wallet: Wallet,
+        chain_id: str,
+        grpc_endpoint: str,
+        credentials: ChannelCredentials = None,
+    ) -> None:
+        if credentials is None:
+            channel = insecure_channel(grpc_endpoint)
+        else:
+            channel = secure_channel(grpc_endpoint, credentials)
```

Comment:
I think this may be better:
`channel = insecure_channel(grpc_endpoint, credentials) if credentials else insecure_channel(grpc_endpoint)` 

or

```
if credentials:
     channel = secure_channel(grpc_endpoint, credentials)
else:
    channel = insecure_channel(grpc_endpoint)
```
or using `if not credentials:`

Reply:
I prefer to explicitly check for `None` to avoid confusion with `False`.\
In this case, it does not matter that much though, as the variable is not a Boolean type.